bin/index.js

```diff
@@ -11,44 +11,76 @@ const jsonFormatter = require('../lib/formatters/json');
 const yamlFormatter = require('../lib/formatters/yaml');
 const SVG = require('../lib/svg');
 
-// Declare the app
-const app = (filePath, options) => {
+const parseStdin = () => {
+  return new Promise((resolve, reject) => {
+    let data = '';
+    
+    process.stdin.on('data', (chunk) => {
+      data += chunk;
+    });
+
+    process.stdin.on('error', (error) => {
+      reject(error);
+    });
+
+    process.stdin.on('end', function() {
+      resolve(data);
+    });
+  });
+};
+
+const parseFile = (filePath) => {
+return new Promise((resolve, reject) => {
   fs.readFile(filePath, (err, data) => {
     if (err) {
-      switch (err.code) {
+      switch(err.code) {
         case 'ENOENT':
-          console.error(
-            `${colors.red('[Error]')} There file ${filePath} doesn't exist`);
+          reject(`${colors.red('[Error]')} There file ${filePath} doesn't exist`);
           break;
         default:
-          console.error(
-            `${colors.red('[Error]')} There was an error opening the ` +
+          reject(`${colors.red('[Error]')} There was an error opening the ` +
             `file: ${err.message}`);
       }
     } else {
-      // Load the SVG
-      try {
+      resolve(data);
+    }
+  });
+});
+};
+
+// Declare the app
+const app = (filePath) => {
+  let dataPromise;
+
+  if (!process.stdin.isTTY) {
+    dataPromise = parseStdin()
+  } else {
+    dataPromise = parseFile(filePath)
+  }
+
+  dataPromise
+    .then((data) => {
         let svg = new SVG(data, filePath);
         let output;
 
-        switch (options.output) {
+        switch (commander.output) {
           case 'json':
-            output = jsonFormatter(svg, options);
+            output = jsonFormatter(svg, commander);
             break;
           case 'yaml':
-            output = yamlFormatter(svg, options);
+            output = yamlFormatter(svg, commander);
             break;
           default:
-            output = humanFormatter(svg, options);
+            if(filePath && !process.stdin.isTTY) {
```

Comment:
Extract this conditional from the `switch`. Maybe we will display a message in the other formatters.